src/test/regress/expected/multi_router_planner.out

```diff
@@ -1505,7 +1505,7 @@ SET citus.enable_non_colocated_router_query_pushdown TO OFF;
 SELECT a.author_id as first_author, b.word_count as second_word_count
 	FROM articles_hash a, articles_single_shard_hash b
 	WHERE a.author_id = 10 and a.author_id = b.author_id and false;
-DEBUG:  router planner does not support queries that reference non-colocated distributed tables
+DEBUG:  Creating router plan
```

Comment:
this creates zero-shard router query.

Reply:
On main branch, when the flag is set to on, the query is multi-shard and when the flag is set to off it is zero shard query.

I guess our code is more mature with zero-shard router queries, in fact I'm a little surprised to see that multi-shard even works.

So, with this patch, zero shard is always router query, which is desirable, I think

```SQL
 set citus.enable_non_colocated_router_query_pushdown to on;
SET
Time: 0.686 ms
[local] onderkalaci@onderkalaci:9700-28622=# SELECT a.author_id as first_author, b.word_count as second_word_count
        FROM articles_hash a, articles_single_shard_hash b
        WHERE a.author_id = 10 and a.author_id = b.author_id and int4eq(1, 2);
DEBUG:  Creating router plan
NOTICE:  executing the command locally: SELECT a.author_id AS first_author, b.word_count AS second_word_count FROM (SELECT NULL::bigint AS id, NULL::bigint AS author_id, NULL::character varying(20) AS title, NULL::integer AS word_count WHERE false) a(id, author_id, title, word_count), (SELECT NULL::bigint AS id, NULL::bigint AS author_id, NULL::character varying(20) AS title, NULL::integer AS word_count WHERE false) b(id, author_id, title, word_count) WHERE ((a.author_id OPERATOR(pg_catalog.=) 10) AND (a.author_id OPERATOR(pg_catalog.=) b.author_id) AND int4eq(1, 2))
┌──────────────┬───────────────────┐
│ first_author │ second_word_count │
├──────────────┼───────────────────┤
└──────────────┴───────────────────┘
(0 rows)

Time: 1.485 ms
[local] onderkalaci@onderkalaci:9700-28622=# set citus.enable_non_colocated_router_query_pushdown to off;
SET
Time: 0.409 ms
[local] onderkalaci@onderkalaci:9700-28622=# SELECT a.author_id as first_author, b.word_count as second_word_count
        FROM articles_hash a, articles_single_shard_hash b
        WHERE a.author_id = 10 and a.author_id = b.author_id and int4eq(1, 2);
DEBUG:  router planner does not support queries that reference non-colocated distributed tables
DEBUG:  combine query: SELECT first_author, second_word_count FROM pg_catalog.citus_extradata_container(10, NULL::cstring(0), NULL::cstring(0), '(i 1 2)'::cstring(0)) remote_scan(first_author bigint, second_word_count integer)
┌──────────────┬───────────────────┐
│ first_author │ second_word_count │
├──────────────┼───────────────────┤
└──────────────┴───────────────────┘
(0 rows)

```